src/jit/morph.cpp

```diff
@@ -5630,7 +5630,8 @@ GenTreePtr Compiler::fgMorphArrayIndex(GenTreePtr tree)
         // dereference.
         // Also we allocate the temporary when the arrRef is sufficiently complex/expensive.
         //
-        if ((arrRef->gtFlags & (GTF_ASG | GTF_CALL | GTF_GLOB_REF)) || gtComplexityExceeds(&arrRef, MAX_ARR_COMPLEXITY))
+        if ((arrRef->gtFlags & (GTF_ASG | GTF_CALL | GTF_GLOB_REF)) ||
```

Comment:
I added a comment.

---
In reply to: [96779213](https://github.com/dotnet/coreclr/pull/8402#discussion_r96779213) [](ancestors = 96779213,95664107)